test/prog.c

```diff
@@ -401,4 +412,58 @@ prog_is_stopped (void)
     return prog_stopped != 0;
 }
 
-
+//Partly taken from https://www.binarytides.com/hostname-to-ip-address-c-sockets-linux/
+int getIpfromDNS(const char* hostname, char* ipaddr, bool version, const char* port)
+{
+	struct addrinfo hints, *servinfo, *p;
+	struct sockaddr_in *h;
+	struct sockaddr_in6 *h6;
+	int rv;
+	char * port2 = strdup(port); /*so that it's no longer const */
+	char * hostname2 = strdup(hostname);
+
+	memset(&hints, 0, sizeof(hints));
+	if (version)
+	{
+		hints.ai_family = AF_INET6;
+	}
+	else
+	{
+		hints.ai_family = AF_INET;
+	}
+	
+	hints.ai_socktype = SOCK_STREAM;
+	
+
+	if ((rv = getaddrinfo(hostname2, port2 , &hints, &servinfo)) != 0)
+	{
+		fprintf(stderr, "getaddrinfo: %s\n", gai_strerror(rv));
+		free(port2);
+		free(hostname2);
+		freeaddrinfo(servinfo);
+		return 1;
+	}
+
+	// loop through all the results and connect to the first we can
+	if (version)/*Ipv6*/
+	{
+		for (p = servinfo; p != NULL; p = p->ai_next)
+		{
+			h6 = (struct sockaddr_in6*) p->ai_addr;
+			inet_ntop(AF_INET6, &h6->sin6_addr, ipaddr, 47);
```

Comment:
47 is an odd-looking bare literal.  Why 47?

Reply:
47 is the size of the ipaddr variable (which is the ip variable we talked about above) and is the maximum size of an ipv6 string. Using sizeof didn't work here since the actual string in the char array can be smaller (in case of ipv4 for example). I'll try to come up with a more elegant way to solve this.

Reply:
At this size, it is safe to allocate this buffer on the stack.